helix-view/src/document.rs

```diff
@@ -633,6 +634,88 @@ impl Document {
         Some(fut.boxed())
     }
 
+    pub fn code_actions_on_save(
+        &self,
+    ) -> Option<BoxFuture<'static, Result<Vec<helix_lsp::lsp::CodeActionOrCommand>, anyhow::Error>>>
+    {
+        let code_actions_on_save = self
+            .language_config()
+            .map(|c| c.code_actions_on_save.clone())?;
+
+        if code_actions_on_save.is_empty() {
+            return None;
+        }
+
+        let request = self.code_actions(self.full_range())?;
+
+        let fut = async move {
+            log::debug!("Configured code actions on save {:?}", code_actions_on_save);
+            let json = request.await?;
+            let response: Option<helix_lsp::lsp::CodeActionResponse> =
+                serde_json::from_value(json)?;
+            let mut code_actions = match response {
+                Some(value) => value,
+                None => helix_lsp::lsp::CodeActionResponse::default(),
+            };
+            log::debug!("Available code actions {:?}", code_actions);
+            code_actions.retain(|action| {
+                matches!(
+                    action,
+                    helix_lsp::lsp::CodeActionOrCommand::CodeAction(x) if x.disabled.is_none() &&
+                        code_actions_on_save.iter().any(|a| match &x.kind {
```

Comment:
probably a good idea to make this a `HashSet` so we avoid being `O(MN)` here even if `M` and `N` is usually small frserializing to a `HashSet` just takes replacing `Vec` with `HashSet`